frontend/controller/service-worker.js

```diff
@@ -194,13 +198,24 @@ sbp('sbp/selectors/register', {
           return subscription
         }).catch(e => {
           if (!(retryCount > 3) && e?.message === 'WebSocket connection is not open') {
-            sbp('okTurtles.events/once', ONLINE, () => {
-              setTimeout(() => sbp('service-worker/setup-push-subscription', (retryCount || 0) + 1), 200)
+            return new Promise((resolve) => {
+              sbp('okTurtles.events/once', ONLINE, () => {
+                setTimeout(() => resolve(sbp('service-worker/setup-push-subscription', (retryCount || 0) + 1)), 200)
+              })
```

Comment:
The way this is written can cause the entire call to `'service-worker/setup-push-subscription'` to hang potentially indefinitely if this event never gets fired.

Can you add a timeout condition to call `reject` on the promise?

Reply:
I've added it as per your request, but the existing behaviour won't result in a ' "stuck" app that just spins forever', since no UI elements depend on this promise.

Reply:
Ah, I see................. maybe that should be a comment then.